src/libstd/io/util.rs

```diff
@@ -155,6 +155,7 @@ pub struct Sink { _priv: () }
 /// let mut buffer = vec![1, 2, 3, 5, 8];
 /// let num_bytes = io::sink().write(&mut buffer).unwrap();
 /// assert_eq!(num_bytes, 5);
+/// assert_eq!(buffer, &[1, 2, 3, 5, 8]);
```

Comment:
Maybe instead of this, just change the example so it doesn't mutably borrow the buffer?